drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-runtime/src/main/java/org/drools/workbench/screens/scenariosimulation/webapp/client/editor/ScenarioSimulationEditorKogitoRuntimeScreen.java

```diff
@@ -0,0 +1,135 @@
+ * entry point. Please note, the <code>IDENTIFIED</code> is the unique key to refer this editor. Please use the same
+ * (eg in kogito-tooling project) to include it in external project or when calling .setContent method (refer to readme.md
+ * for further information */
+@ApplicationScoped
+@WorkbenchClientEditor(identifier = IDENTIFIER)
+public class ScenarioSimulationEditorKogitoRuntimeScreen extends AbstractScenarioSimulationEditorKogitoScreen {
+
+    protected static final String NEW_FILE_NAME = "new-file.scesim";
+    protected static final PlaceRequest SCENARIO_SIMULATION_KOGITO_RUNTIME_SCREEN_DEFAULT_REQUEST = new DefaultPlaceRequest(IDENTIFIER);
+
+    protected PlaceManager placeManager;
+
+    public ScenarioSimulationEditorKogitoRuntimeScreen() {
+        //CDI proxy
+    }
+
+    @Inject
+    public ScenarioSimulationEditorKogitoRuntimeScreen(final PlaceManager placeManager) {
+        this.placeManager = placeManager;
+    }
+
+    @Override
+    public PlaceRequest getPlaceRequest() {
+        return SCENARIO_SIMULATION_KOGITO_RUNTIME_SCREEN_DEFAULT_REQUEST;
+    }
+
+    @OnStartup
+    public void onStartup(final PlaceRequest place) {
+        scenarioSimulationEditorKogitoWrapper.onStartup(place);
+    }
+
+    @OnMayClose
+    public boolean mayClose() {
+        return scenarioSimulationEditorKogitoWrapper.mayClose();
+    }
+
+    @WorkbenchPartTitle
+    public String getTitleText() {
+        return TITLE;
+    }
+
+    @WorkbenchPartTitleDecoration
+    public IsWidget getTitle() {
+        return scenarioSimulationEditorKogitoWrapper.getTitle();
+    }
+
+    @WorkbenchPartView
+    public MultiPageEditorContainerView getWidget() {
+        return scenarioSimulationEditorKogitoWrapper.getWidget();
+    }
+
+    @WorkbenchMenu
+    public void setMenus(final Consumer<Menus> menusConsumer) {
+        scenarioSimulationEditorKogitoWrapper.setMenus(menusConsumer);
+    }
+
+    @GetContent
+    public Promise getContent() {
+        return scenarioSimulationEditorKogitoWrapper.getContent();
+    }
+
+    @SetContent
+    public void setContent(String fullPath, String value) {
+        /* Retrieving file name and its related path */
+        String finalName = NEW_FILE_NAME;
+        String pathString = "/";
+        if (fullPath != null && !fullPath.isEmpty()) {
+            int idx = fullPath.replaceAll("\\\\", "/").lastIndexOf('/');
+            finalName = idx >= 0 ? fullPath.substring(idx + 1) : fullPath;
+            pathString = idx >= 0 ? fullPath.substring(0, idx + 1) : "";
+        }
+        final Path path = PathFactory.newPath(finalName, pathString);
+
+        if (value == null || value.isEmpty()) {
+            newFile(path);
+        } else {
+            scenarioSimulationEditorKogitoWrapper.gotoPath(path);
+            scenarioSimulationEditorKogitoWrapper.setContent(path.toURI() + path.getFileName(), value);
+            scenarioKogitoCreationPopupPresenter.hide();
+        }
+    }
+
+    @IsDirty
```

Comment:
Idea suggests to use `OnMayClose` as `IsDirty` is deprecated.

Reply:
@jomarko I saw that, but in both DMNEditor and BPMN `@IsDirty` is used. So, I would prefer to follow the same approach. In case, we can evaluate to change it for all editors, if is correct.

Reply:
ok